src/EnergyPlus/UnitVentilator.cc

```diff
@@ -804,469 +650,364 @@ namespace UnitVentilator {
                 if (!lAlphaBlanks(18)) {
-                    state.dataUnitVentilators->UnitVent(UnitVentNum).CCoilPresent = true;
+                    unitVent.CCoilPresent = true;
                     errFlag = false;
 
                     cCoolingCoilType = Alphas(17);
-                    state.dataUnitVentilators->UnitVent(UnitVentNum).CCoilTypeCh = cCoolingCoilType;
+                    unitVent.CCoilTypeCh = cCoolingCoilType;
                     {
-                        auto const SELECT_CASE_var(cCoolingCoilType);
-                        if (SELECT_CASE_var == "COIL:COOLING:WATER") {
-                            state.dataUnitVentilators->UnitVent(UnitVentNum).CCoilType = Cooling_CoilWaterCooling;
-                            state.dataUnitVentilators->UnitVent(UnitVentNum).CoolingCoilType = DataPlant::PlantEquipmentType::CoilWaterCooling;
-                            state.dataUnitVentilators->UnitVent(UnitVentNum).CCoilPlantName = Alphas(18);
-                        } else if (SELECT_CASE_var == "COIL:COOLING:WATER:DETAILEDGEOMETRY") {
-                            state.dataUnitVentilators->UnitVent(UnitVentNum).CCoilType = Cooling_CoilDetailedCooling;
-                            state.dataUnitVentilators->UnitVent(UnitVentNum).CoolingCoilType =
-                                DataPlant::PlantEquipmentType::CoilWaterDetailedFlatCooling;
-                            state.dataUnitVentilators->UnitVent(UnitVentNum).CCoilPlantName = Alphas(18);
-                        } else if (SELECT_CASE_var == "COILSYSTEM:COOLING:WATER:HEATEXCHANGERASSISTED") {
-                            state.dataUnitVentilators->UnitVent(UnitVentNum).CCoilType = Cooling_CoilHXAssisted;
-                            GetHXCoilTypeAndName(state,
-                                                 cCoolingCoilType,
-                                                 Alphas(18),
-                                                 ErrorsFound,
-                                                 state.dataUnitVentilators->UnitVent(UnitVentNum).CCoilPlantType,
-                                                 state.dataUnitVentilators->UnitVent(UnitVentNum).CCoilPlantName);
-                            if (UtilityRoutines::SameString(state.dataUnitVentilators->UnitVent(UnitVentNum).CCoilPlantType, "Coil:Cooling:Water")) {
-                                state.dataUnitVentilators->UnitVent(UnitVentNum).CoolingCoilType = DataPlant::PlantEquipmentType::CoilWaterCooling;
-                            } else if (UtilityRoutines::SameString(state.dataUnitVentilators->UnitVent(UnitVentNum).CCoilPlantType,
-                                                                   "Coil:Cooling:Water:DetailedGeometry")) {
-                                state.dataUnitVentilators->UnitVent(UnitVentNum).CoolingCoilType =
-                                    DataPlant::PlantEquipmentType::CoilWaterDetailedFlatCooling;
+                        auto const coolingCoilType(cCoolingCoilType);
```

Comment:
Same comment as above. BTW, really happy to see many of these changes! Code starting to look much more C++-like.